### ISSUE TITLE
[WAGON-521] Maven cannot use system default proxies

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-wagon#579]
**[Noel Grandin](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=grandinj)** opened **[WAGON-521](https://issues.apache.org/jira/browse/WAGON-521?redirect=false)** and commented

when using maven from behind a proxy, it cannot connect to download

The maven/java stuff will not use the system proxies unless explicitly told to do so, which is a problem when maven tries to download the jars it needs :-)

Basically, I need the following option added to the java command line:
-Djava.net.useSystemProxies=true

Sun did not see to make this the default because of backwards compatibility, which is a right PITA, but anyhow :-)

Please also see original issue here:
https://github.com/takari/maven-wrapper/issues/58
https://github.com/pmd/pmd/issues/498

---

**Affects:** 3.1.0

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

It won't do anyway because this variable is used by `URLConnection`. Maven uses Apache HttpClient for transport by default. It has its own proxy config.

**Comment by jira-importer** [comment]
**[Noel Grandin](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=grandinj)** commented

Looks like Apache HttpClient has a mechanism for correctly picking up system settings, provided HttpClient it itself configured correctly?
SystemDefaultHttpClient
https://issues.apache.org/jira/browse/HTTPCLIENT-1128

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

This is by default not activated. You can investigate and we can move this to Wagon.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

No investigation nor a patch have been provided.

**Comment by jira-importer** [comment]
**[Noel Grandin](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=grandinj)** commented

So maven operates on the principle that unless a patch is provided a bug does not exist?

That's a little sad

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Well, the issue was closed with the wrapper too. I have offered a year ago: You can investigate and we can move this to Wagon.

In fact, this property works already by default in HttpClient for years. You just don't have to provide any proxy in your `settings.xml`. Have you tried that at least?

**Comment by jira-importer** [comment]
**[Noel Grandin](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=grandinj)** commented

I am sorry, I don't know what Wagon is. I assume you mean some internal component of Maven?

On Windows, by default, out of the box, Maven does not use the system configured proxy. This is with an empty settings.xml.

By "system proxy" I mean the proxy configured globally, in Windows Control Panel.

I logged the bug here because mvnwrapper said they thought this should be fixed in core Maven, or whatever part of core Maven handles proxies.

Does that help?

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Wagon handles the transport passing to HttpClient. Maven will not do this by default because the default value of `java.net.useSystemProxies` is `false`. If you set to `true` it will work. There is nothing to fix here nor is there a bug. Leave your settings empty and pass `true`.

At most what I would love to see is a vastly improved proxy support which does nor than defining a static list of proxy. Pretty much like browsers provide it these days.
